tests/nodescaling/scale_test.go

```diff
@@ -0,0 +1,157 @@
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/users"
+	password "github.com/rancher/rancher/tests/framework/extensions/users/passwordgenerator"
+	ranchFrame "github.com/rancher/rancher/tests/framework/pkg/config"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/tfp-automation/config"
+	"github.com/rancher/tfp-automation/framework"
+	cleanup "github.com/rancher/tfp-automation/framework/cleanup"
+	"github.com/rancher/tfp-automation/tests/extensions/provisioning"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ScaleTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	terraformConfig    *config.TerraformConfig
+	clusterConfig      *config.TerratestConfig
+	terraformOptions   *terraform.Options
+}
+
+func (r *ScaleTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	terraformConfig := new(config.TerraformConfig)
+	ranchFrame.LoadConfig("terraform", terraformConfig)
+
+	r.terraformConfig = terraformConfig
+
+	clusterConfig := new(config.TerratestConfig)
+	ranchFrame.LoadConfig("terratest", clusterConfig)
+
+	r.clusterConfig = clusterConfig
+
+	terraformOptions, err := framework.Setup(r.T())
+	require.NoError(r.T(), err)
+
+	r.terraformOptions = terraformOptions
+}
+
+func (r *ScaleTestSuite) TestScale() {
+	nodeRolesAll := []config.Nodepool{config.AllRolesNodePool}
+	newWorkerAll := []config.Nodepool{config.AllRolesNodePool, config.WorkerNodePool}
+	nodeRolesShared := []config.Nodepool{config.EtcdControlPlaneNodePool, config.WorkerNodePool}
+	newWorkersShared := []config.Nodepool{config.EtcdControlPlaneNodePool, config.TwoWorkersNodePool}
+	nodeRolesDedicated := []config.Nodepool{config.EtcdNodePool, config.ControlPlaneNodePool, config.WorkerNodePool}
+	newWorkersDedicated := []config.Nodepool{config.EtcdNodePool, config.ControlPlaneNodePool, config.TwoWorkersNodePool}
+
+	tests := []struct {
+		name            string
+		nodeRoles       []config.Nodepool
+		scaledNodeRoles []config.Nodepool
+		client          *rancher.Client
```

Comment:
shouldn't ScaledUpNode and ScaleDownNode count be hard coded in the static tests? if not, why?

Reply:
That isn't necessary to do that, we only need to ensure that we are scaling the nodeRoles in the test cases. The node count and node pools scaling can be handled directly after, which is what we're doing.